### PR TITLE
[KYUUBI #3456] Ldap uses unboundid-ldapsdk to replace the original java implementation

### LINKED ISSUE #3456: [Improvement] Ldap uses unboundid-ldapsdk to replace the original java implementation
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

 It offers better performance, better ease of use, and more features than other Java-based LDAP APIs.

### How should we improve?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by jiaoqingbo**
### _Why are the changes needed?_

fix #3456 

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by jiaoqingbo** [comment]
The failed test seems not related to this pr

File: kyuubi-common/src/main/scala/org/apache/kyuubi/service/authentication/LdapAuthenticationProviderImpl.scala [hunk 1]
```diff
@@ -41,47 +44,119 @@ class LdapAuthenticationProviderImpl(conf: KyuubiConf) extends PasswdAuthenticat
    * @throws AuthenticationException When a user is found to be invalid by the implementation
    */
   override def authenticate(user: String, password: String): Unit = {
-    if (StringUtils.isBlank(user)) {
+
+    var bindUser = conf.get(AUTHENTICATION_LDAP_BIND_USER).getOrElse("")
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/service/authentication/LdapAuthenticationProviderImpl.scala:48]: can we extract lines 48 to 71 to a method? e.g. 
```
def resolveBindUserPassword(): (String, String)
```

File: kyuubi-common/src/main/scala/org/apache/kyuubi/service/authentication/LdapAuthenticationProviderImpl.scala [hunk 2]
```diff
@@ -41,47 +44,119 @@ class LdapAuthenticationProviderImpl(conf: KyuubiConf) extends PasswdAuthenticat
...
+
+  /**
+   * Reads and parses DN patterns from Kyuubi configuration.
+   * <br>
+   * If no patterns are provided in the configuration, then the base DN will be used.
+   *
+   * @param conf Kyuubi configuration
+   * @return a list of DN patterns
+   */
+  private def parseDnPatterns(conf: KyuubiConf): util.List[String] = {
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/service/authentication/LdapAuthenticationProviderImpl.scala:146]: prefer to use Array or scala collections

File: kyuubi-common/src/main/scala/org/apache/kyuubi/service/authentication/LdapAuthenticationProviderImpl.scala [hunk 3]
```diff
@@ -41,47 +44,119 @@ class LdapAuthenticationProviderImpl(conf: KyuubiConf) extends PasswdAuthenticat
...
+  private def isDn(name: String): Boolean = name.contains("=")
+
+  /**
+   * Creates a list of principals to be used for user authentication.
+   *
+   * @param conf Kyuubi configuration
+   * @param user username
+   * @return a list of user's principals
+   */
+  private def createCandidatePrincipals(conf: KyuubiConf, user: String): util.List[String] = {
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/service/authentication/LdapAuthenticationProviderImpl.scala:123]: prefer to use Array or scala collections

**Pull request closed without merging**